inference-engine/src/mkldnn_plugin/nodes/mkldnn_shuffle_channels_node.cpp

```diff
@@ -0,0 +1,843 @@
+#define GET_SHUFFLE_INDEX(idx) (idx * group_size % shuffle_size + idx * group_size / shuffle_size)
+#define GET_PTR(in_work_offset, out_work_offset, work_amount) const uint8_t *in_p  = in_ptr  +  (in_work_offset) * (work_amount) * data_size; \
+                                                                    uint8_t *out_p = out_ptr + (out_work_offset) * (work_amount) * data_size
+
+// some utility functions
+static inline bool isFloatCompatible(memory::data_type type) {
+    return memory::f32 == type || memory::bf16 == type;
+}
+
+template <cpu_isa_t isa>
+struct jit_uni_shuffle_channels_kernel_f32 : public jit_uni_shuffle_channels_kernel, public jit_generator {
+    DECLARE_CPU_JIT_AUX_FUNCTIONS(jit_uni_shuffle_channels_kernel_f32)
+
+    explicit jit_uni_shuffle_channels_kernel_f32(jit_shuffle_channels_config_params jcp)
+    : jit_uni_shuffle_channels_kernel(jcp), jit_generator() {
+        if (!mayiuse(avx512_core_bf16) && mayiuse(avx512_core))
+            emu_vcvtneps2bf16.reset(new jit_emu_vcvtneps2bf16(this, isa, nullptr));
+
+        this->preamble();
+
+        mov(reg_src, ptr[reg_params + GET_OFF(src)]);
+        mov(reg_dst, ptr[reg_params + GET_OFF(dst)]);
+        mov(reg_index, ptr[reg_params + GET_OFF(index)]);
+        mov(reg_work_amount, ptr[reg_params + GET_OFF(work_amount)]);
+
+        if (jcp_.shuffle_innermost) {
+            shuffle_loop_gather();
+        } else {
+            shuffle_loop();
+        }
+
+        this->postamble();
+
+        if (!mayiuse(avx512_core_bf16) && mayiuse(avx512_core))
+            emu_vcvtneps2bf16->emit_table();
+
+        ker_ = (decltype(ker_)) this->getCode();
+    }
+
+private:
+    using Vmm = typename conditional3<isa == cpu::sse42, Xbyak::Xmm, isa == cpu::avx2,
+            Xbyak::Ymm, Xbyak::Zmm>::type;
+    size_t vlen = cpu_isa_traits<isa>::vlen;
+
+    Xbyak::Reg64 reg_src = r8;
+    Xbyak::Reg64 reg_dst = r9;
+    Xbyak::Reg64 reg_index = r10;
+    Xbyak::Reg64 reg_work_amount = r11;
+    Xbyak::Reg64 reg_params = abi_param1;
+
+    Xbyak::Reg8 reg_tmp_8 = r12b;
+    Xbyak::Reg32 reg_tmp_32 = r12d;
+    Xbyak::Reg64 reg_tmp_64 = r12;
+
+    Vmm vmm_val = Vmm(0);
+    Xmm xmm_val = Xmm(0);
+    Vmm vmm_index = Vmm(1);
+    Xmm xmm_index = Xmm(1);
+    Vmm vmm_mask = Vmm(2);
+    Xmm xmm_mask = Xmm(2);
+    Vmm vmm_zero = Vmm(3);
+
+    const Xbyak::Opmask k_mask = Xbyak::Opmask(1);
+
+    std::unique_ptr<jit_emu_vcvtneps2bf16> emu_vcvtneps2bf16;
+
+    inline void shuffle_loop() {
+        Xbyak::Label shuffle_main_loop_label;
+        Xbyak::Label shuffle_main_loop_end_label;
+        int step = vlen / sizeof(float) < 8 ? 8 : vlen / sizeof(float);
+        L(shuffle_main_loop_label);
+        {
+            cmp(reg_work_amount, step);
+            jl(shuffle_main_loop_end_label, T_NEAR);
+
+            load_vector(vmm_val, ptr[reg_src], jcp_.data_type);
+            store_vector(ptr[reg_dst], vmm_val, jcp_.data_type);
```

Comment:
As I can see `load_vector` and `store_vector` routines performs conversion from src data type to fp32 during load and from fp32 to dst data type during store. Why we need this? ShuffleChannels is data type agnostic operation and just manipulates defined number of bytes. I expect there shouldn't be any data type conversions at all.